unconfigured-node-name fails with assertion error on windows

https://scans.gradle.com/s/rleubfn3somoa/console-log?task=:qa:unconfigured-node-name:integTestRunner#L34

```
Â» [2019-07-20T09:42:25,054][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [] fatal error in thread [main], exiting
--
Â»  java.lang.AssertionError: null
Â»  	at org.elasticsearch.common.util.concurrent.EsExecutors.daemonThreadFactory(EsExecutors.java:238) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Â»  	at org.elasticsearch.cluster.service.ClusterApplierService.createThreadPoolExecutor(ClusterApplierService.java:142) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Â»  	at org.elasticsearch.cluster.service.ClusterApplierService.doStart(ClusterApplierService.java:136) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Â»  	at org.elasticsearch.common.component.AbstractLifecycleComponent.start(AbstractLifecycleComponent.java:59) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Â»  	at org.elasticsearch.cluster.service.ClusterService.doStart(ClusterService.java:101) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Â»  	at org.elasticsearch.common.component.AbstractLifecycleComponent.start(AbstractLifecycleComponent.java:59) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Â»  	at org.elasticsearch.node.Node.start(Node.java:706) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Â»  	at org.elasticsearch.bootstrap.Bootstrap.start(Bootstrap.java:271) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Â»  	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:348) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Â»  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:159) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Â»  	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:150) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Â»  	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Â»  	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:124) ~[elasticsearch-cli-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Â»  	at org.elasticsearch.cli.Command.main(Command.java:90) ~[elasticsearch-cli-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Â»  	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:115) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
Â»  	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:92) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
```

--- a/buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java
+++ b/buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java
@@ -107,6 +107,8 @@ public class ElasticsearchNode implements TestClusterConfiguration {
         "is a pre-release version of Elasticsearch",
         "max virtual memory areas vm.max_map_count"
     );
+    private static final String LINUX_DARWIN_HOSTNAME_OVERRIDE = "LinuxDarwinHostname";
+    private static final String WINDOWS_COMPUTERNAME_OVERRIDE = "WindowsComputername";
 
     private final String path;
     private final String name;
@@ -642,10 +644,10 @@ public class ElasticsearchNode implements TestClusterConfiguration {
         defaultEnv.put("ES_TMPDIR", tmpDir.toString());
         // Windows requires this as it defaults to `c:\windows` despite ES_TMPDIR
         defaultEnv.put("TMP", tmpDir.toString());
-        // Windows needs the COMPUTERNAME variable in order to set HOSTNAME
-        if (System.getenv("COMPUTERNAME") != null) {
-            defaultEnv.put("COMPUTERNAME", System.getenv("COMPUTERNAME"));
-        }
+
+        // Override the system hostname variables for testing
+        defaultEnv.put("HOSTNAME", LINUX_DARWIN_HOSTNAME_OVERRIDE);
+        defaultEnv.put("COMPUTERNAME", WINDOWS_COMPUTERNAME_OVERRIDE);
 
         Set<String> commonKeys = new HashSet<>(environment.keySet());
         commonKeys.retainAll(defaultEnv.keySet());

--- a/buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java
+++ b/buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java
@@ -642,6 +642,8 @@ public class ElasticsearchNode implements TestClusterConfiguration {
         defaultEnv.put("ES_TMPDIR", tmpDir.toString());
         // Windows requires this as it defaults to `c:\windows` despite ES_TMPDIR
         defaultEnv.put("TMP", tmpDir.toString());
+        // Windows needs the COMPUTERNAME variable in order to set HOSTNAME
+        defaultEnv.put("COMPUTERNAME", System.getenv("COMPUTERNAME"));
 
         Set<String> commonKeys = new HashSet<>(environment.keySet());
         commonKeys.retainAll(defaultEnv.keySet());

--- a/buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java
+++ b/buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java
@@ -107,8 +107,8 @@ public class ElasticsearchNode implements TestClusterConfiguration {
         "is a pre-release version of Elasticsearch",
         "max virtual memory areas vm.max_map_count"
     );
-    private static final String LINUX_DARWIN_HOSTNAME_OVERRIDE = "LinuxDarwinHostname";
-    private static final String WINDOWS_COMPUTERNAME_OVERRIDE = "WindowsComputername";
+    private static final String HOSTNAME_OVERRIDE = "LinuxDarwinHostname";
+    private static final String COMPUTERNAME_OVERRIDE = "WindowsComputername";
 
     private final String path;
     private final String name;
@@ -646,8 +646,8 @@ public class ElasticsearchNode implements TestClusterConfiguration {
         defaultEnv.put("TMP", tmpDir.toString());
 
         // Override the system hostname variables for testing
-        defaultEnv.put("HOSTNAME", LINUX_DARWIN_HOSTNAME_OVERRIDE);
-        defaultEnv.put("COMPUTERNAME", WINDOWS_COMPUTERNAME_OVERRIDE);
+        defaultEnv.put("HOSTNAME", HOSTNAME_OVERRIDE);
+        defaultEnv.put("COMPUTERNAME", COMPUTERNAME_OVERRIDE);
 
         Set<String> commonKeys = new HashSet<>(environment.keySet());
         commonKeys.retainAll(defaultEnv.keySet());

--- a/buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java
+++ b/buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java
@@ -643,7 +643,9 @@ public class ElasticsearchNode implements TestClusterConfiguration {
         // Windows requires this as it defaults to `c:\windows` despite ES_TMPDIR
         defaultEnv.put("TMP", tmpDir.toString());
         // Windows needs the COMPUTERNAME variable in order to set HOSTNAME
-        defaultEnv.put("COMPUTERNAME", System.getenv("COMPUTERNAME"));
+        if (System.getenv("COMPUTERNAME") != null) {
+            defaultEnv.put("COMPUTERNAME", System.getenv("COMPUTERNAME"));
+        }
 
         Set<String> commonKeys = new HashSet<>(environment.keySet());
         commonKeys.retainAll(defaultEnv.keySet());

--- a/qa/unconfigured-node-name/build.gradle
+++ b/qa/unconfigured-node-name/build.gradle
@@ -30,6 +30,4 @@ testClusters.integTest {
 integTest.runner {
   nonInputProperties.systemProperty 'tests.logfile',
     "${ -> testClusters.integTest.singleNode().getServerLog() }"
-  // https://github.com/elastic/elasticsearch/issues/44656
-  onlyIf { OS.WINDOWS.equals(OS.current()) == false }
 }

--- a/qa/unconfigured-node-name/src/test/java/org/elasticsearch/unconfigured_node_name/JsonLogsFormatAndParseIT.java
+++ b/qa/unconfigured-node-name/src/test/java/org/elasticsearch/unconfigured_node_name/JsonLogsFormatAndParseIT.java
@@ -37,15 +37,15 @@ public class JsonLogsFormatAndParseIT extends JsonLogsIntegTestCase {
     private static final boolean WINDOWS = OS_NAME.startsWith("Windows");
 
     // These match the values defined in org.elasticsearch.gradle.testclusters.ElasticsearchNode
-    private static final String WINDOWS_COMPUTERNAME = "WindowsComputername";
-    private static final String LINUX_DARWIN_HOSTNAME = "LinuxDarwinHostname";
+    private static final String COMPUTERNAME = "WindowsComputername";
+    private static final String HOSTNAME = "LinuxDarwinHostname";
 
     @Override
     protected Matcher<String> nodeNameMatcher() {
         if (WINDOWS) {
-            return equalTo(WINDOWS_COMPUTERNAME);
+            return equalTo(COMPUTERNAME);
         }
-        return equalTo(LINUX_DARWIN_HOSTNAME);
+        return equalTo(HOSTNAME);
     }
 
     @Override

--- a/qa/unconfigured-node-name/src/test/java/org/elasticsearch/unconfigured_node_name/JsonLogsFormatAndParseIT.java
+++ b/qa/unconfigured-node-name/src/test/java/org/elasticsearch/unconfigured_node_name/JsonLogsFormatAndParseIT.java
@@ -30,12 +30,22 @@ import java.nio.file.Path;
 import java.security.AccessController;
 import java.security.PrivilegedAction;
 
-import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.equalTo;
 
 public class JsonLogsFormatAndParseIT extends JsonLogsIntegTestCase {
+    private static final String OS_NAME = System.getProperty("os.name");
+    private static final boolean WINDOWS = OS_NAME.startsWith("Windows");
+
+    // These match the values defined in org.elasticsearch.gradle.testclusters.ElasticsearchNode
+    private static final String WINDOWS_COMPUTERNAME = "WindowsComputername";
+    private static final String LINUX_DARWIN_HOSTNAME = "LinuxDarwinHostname";
+
     @Override
     protected Matcher<String> nodeNameMatcher() {
-        return not("");
+        if (WINDOWS) {
+            return equalTo(WINDOWS_COMPUTERNAME);
+        }
+        return equalTo(LINUX_DARWIN_HOSTNAME);
     }
 
     @Override

